agent/handlers/v2/response.go

```diff
@@ -272,3 +288,34 @@ func NewContainerResponse(
 	resp.Volumes = v1.NewVolumesResponse(dockerContainer)
 	return resp
 }
+
+// metadataErrorHandling writes an error to the logger, and append an error response
+// to V4 metadata endpoint task response
+func metadataErrorHandling(resp *TaskResponse, err error, field, resourceARN string, includeV4Metadata bool) {
+	seelog.Errorf("V2 response: unable to get '%s' for '%s': %s", field, resourceARN, err.Error())
+	if includeV4Metadata {
+		errResp, err := newErrorResponse(err, field, resourceARN)
+		if err == nil {
+			resp.Errors = append(resp.Errors, *errResp)
+		}
+	}
+}
+
+// newErrorResponse creates a new error response
+func newErrorResponse(err error, field, resourceARN string) (*ErrorResponse, error) {
+	if awsErr, ok := err.(awserr.Error); ok {
+		if reqErr, ok := err.(awserr.RequestFailure); ok {
+			errResp := &ErrorResponse{
```

Comment:
nit: no need for the `errResp` variable, you can return `&ErrorResponse{...}` directly